catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/CreateOperations.java

```diff
@@ -117,96 +125,29 @@ public void setHistorian(Historian historian) {
-
-        validateCreateRequest(createRequest);
-
-        if (Requests.isLocal(createRequest)
-                && !sourceOperations.isSourceAvailable(sourceOperations.getCatalog())) {
-            SourceUnavailableException sourceUnavailableException = new SourceUnavailableException(
-                    "Local provider is not available, cannot perform create operation.");
-            if (INGEST_LOGGER.isWarnEnabled()) {
-                INGEST_LOGGER.warn(
-                        "Error on create operation, local provider not available. {} metacards failed to ingest. {}",
-                        createRequest.getMetacards()
-                                .size(),
-                        buildIngestLog(createRequest),
-                        sourceUnavailableException);
-            }
-            throw sourceUnavailableException;
-        }
-
         CreateResponse createResponse = null;
 
         Exception ingestError = null;
-        try {
-            createRequest = injectAttributes(createRequest);
-
-            setDefaultValues(createRequest);
-
-            Map<String, Serializable> unmodifiablePropertiesMap = Collections.unmodifiableMap(
-                    createRequest.getProperties());
-            HashMap<String, Set<String>> requestPolicyMap = new HashMap<>();
-            for (Metacard metacard : createRequest.getMetacards()) {
-                HashMap<String, Set<String>> itemPolicyMap = new HashMap<>();
-                for (PolicyPlugin plugin : frameworkProperties.getPolicyPlugins()) {
-                    PolicyResponse policyResponse = plugin.processPreCreate(metacard,
-                            unmodifiablePropertiesMap);
-                    opsSecuritySupport.buildPolicyMap(itemPolicyMap,
-                            policyResponse.itemPolicy()
-                                    .entrySet());
-                    opsSecuritySupport.buildPolicyMap(requestPolicyMap,
-                            policyResponse.operationPolicy()
-                                    .entrySet());
-                }
 
-                metacard.setAttribute(new AttributeImpl(Metacard.SECURITY, itemPolicyMap));
-            }
-            createRequest.getProperties()
-                    .put(PolicyPlugin.OPERATION_SECURITY, requestPolicyMap);
+        createRequest = queryOperations.setFlagsOnRequest(createRequest);
+        createRequest = validateCreateRequest(createRequest);
+        createRequest = validateLocalSource(createRequest);
 
-            for (AccessPlugin plugin : frameworkProperties.getAccessPlugins()) {
-                createRequest = plugin.processPreCreate(createRequest);
-            }
+        try {
+            createRequest = injectAttributes(createRequest);
+            createRequest = setDefaultValues(createRequest);
+            createRequest = updateCreateRequestPolicyMap(createRequest);
+            createRequest = processPrecreateAccessPlugins(createRequest);
```

Comment:
I guess my hope here would be that, by creating smaller, simpler classes/lambdas, we would be able to extract the similarities between the different operation classes, make them more generic and possibly re-use some of those smaller classes across operations, if that makes sense. I'm not sure that moving code to private methods will allow us to achieve the same thing.

My concern I guess is that if we don't apply some of those patterns anywhere we won't be able to extract as much common code and reduce the amount of repeated code and/or logic as much as we should.

I'm fine waiting for another iteration to try some of this out though. Doesn't have to be part of this PR.